exec/exec.go

```diff
@@ -12,12 +12,14 @@ import (
 // dcos-go/exec is a os/exec wrapper. It implements io.Reader and can be used to read both STDOUT and STDERR.
 //
 // Usage:
-// ce := exec.Run("bash", []string{"infinite.sh"}, exec.Timeout(3 * time.Second))
+// ctx, cancel := context.WithTimeout(context.Background(), 3*time.Second)
+// defer cancel()
 //
+// ce, _ := exec.Run(ctx, "bash", []string{"infinite.sh"})
```

Comment:
Yeah, returning the error is part of the `exec.Run` API, so we should handle it in the example. I also wondered why the API returned an error but the code did not and I'm guessing that you're right that it was left there for later implementation.